include/pms-utils/misc/try_parse.hpp

```diff
@@ -0,0 +1,25 @@
+#include "boost/parser/parser.hpp"
+
+#include <expected>
+#include <iterator>
+#include <string_view>
+
+namespace [[gnu::visibility("default")]] pms_utils {
+namespace misc {
+template <typename Rule, std::ranges::range Range>
+
+std::expected<typename Rule::parser_type::attr_type, typename Range::iterator>
+try_parse(const Range &input, const Rule &rule, bool partial) {
```

Comment:
has this been clang-formatted? the newline between the template and function prototype looks wrong